srai/joiners/base.py

```diff
@@ -0,0 +1,31 @@
+"""Base class for joiners."""
+
+import abc
+
+import geopandas as gpd
+
+from ..regionizers.base import BaseRegionizer
+
+
+class BaseJoiner(abc.ABC):
+    """Base abstract class for joiners."""
+
+    def __init__(self, regionizer: BaseRegionizer):
+        """Initialize the joiner.
+
+        Args:
+            regionizer: Regionizer to be used.
+        """
+        self.regionizer = regionizer
+
+    @abc.abstractmethod
+    def join(self, gdf: gpd.GeoDataFrame) -> gpd.GeoDataFrame:
```

Comment:
Is this GeoDataFrame data from the Loader or raw data to pass to Regionizer? Maybe we should cache results in the Regionizer so we can simply use it in this stage internally.
The second option would be passing both the Loader and Regionizer to ensure that passed GDFs are in a proper format but that seems really limiting for an end user.

Reply:
Does it make any differenece? Regionizer's job is to map any geometry inside a gdf to a *region*, and Joiner's is to group by *region* and apply proper aggregation on other features. I don't think there will be format issue here, but correct me if I'm wrong

Reply:
So are we expecting to send the same GDF into both Loader and Regionizer?
I'd expect that one GDF contains generated regions and the second one contains any geometries that are expected to be joined using spatial operations (intersections or other).

Reply:
I guess those are just two approaches. I went with the one which expects to call `Regionizer` inside the `Joiner`, but we can also use a spatial join here (then caching results in `Regionizer` will be needed)